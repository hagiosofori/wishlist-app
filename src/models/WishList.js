import { types } from "mobx-state-tree"

export const WishListItem = types
	.model({
		name: types.string,
		price: types.number,
		image: "",
	})
	.actions(self => ({
		setName(newName) {
			self.name = newName
		},
		setPrice(newPrice) {
			self.price = newPrice
		},
	}))

export const WishList = types
	.model({
		items: types.optional(types.array(WishListItem), []),
	})
	.actions(self => ({
		addItem(newItem) {
			self.items.push(newItem)
		},
	}))
