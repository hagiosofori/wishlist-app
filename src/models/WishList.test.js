import { WishListItem, WishList } from "./WishList"
import { getSnapshot } from "mobx-state-tree"

it("can create an instance of WishList item", () => {
	//create an item, and then check if the created item is an instance of wishlist, and it has the correct properties
	//also test if the image defaults to an empty string
	const item = WishListItem.create({
		name: "Test Item Name",
		price: 10,
	})

	expect(item.price).toBe(10)
	expect(item.image).toBe("")

	item.setPrice(25)

	expect(item.price).toBe(25)
})

it("can set name of a WishList item", () => {
	const initialItem = WishListItem.create({ name: "Name 1", price: 10 })
	initialItem.setName("Name 2")

	expect(initialItem.name).toBe("Name 2")
})

it("can create an instance of WishList", () => {
	const item = WishList.create({
		items: [
			{
				name: "Test item 1",
				price: 10,
			},
			{ name: "Test item 2", price: 15 },
		],
	})

	expect(item.items.length).toBe(2)
	expect(item.items[0].price).toBe(10)
	expect(item.items[1].image).toBe("")

	item.addItem(WishListItem.create({ name: "test item 3", price: 25 }))
	expect(item.items.length).toBe(3)
	expect(item.items[2].name).toBe("test item 3")

	item.items[2].setName("test name 3")
	expect(item.items[2].name).toBe("test name 3")

	expect(getSnapshot(item)).toMatchSnapshot()
})
