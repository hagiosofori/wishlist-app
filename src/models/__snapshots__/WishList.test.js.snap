// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can create an instance of WishList 1`] = `
Object {
  "items": Array [
    Object {
      "image": "",
      "name": "Test item 1",
      "price": 10,
    },
    Object {
      "image": "",
      "name": "Test item 2",
      "price": 15,
    },
    Object {
      "image": "",
      "name": "test name 3",
      "price": 25,
    },
  ],
}
`;
